src/chart/candlestick/candlestickLayout.js

```diff
@@ -195,11 +195,7 @@ function getSign(data, dataIndex, openVal, closeVal, closeDim) {
         sign = 1;
     }
     else {
-        sign = dataIndex > 0
-            // If close === open, compare with close of last record
-            ? (data.get(closeDim, dataIndex - 1) <= closeVal ? 1 : -1)
-            // No record of previous, set to be positive
-            : 1;
+        sign = 0;
```

Comment:
I don't think you should delete this logic because there are scenarios (see #6583) where doji is expected to use `color0` or `color1` depending on previous values. Current logic is https://github.com/apache/incubator-echarts/issues/6583#issuecomment-326882236.

I'd suggest providing a new option like `useColor2ForDoji` for choosing whether to use `color2`. The default value should be `false` to compat with the old behavior.

Reply:
this scenarios may occurs in limit at the opening(开盘即涨停或开盘即跌停), it means the value of open close highest lowest are the same, but doji's highest and lowest are not the same, I will modify the logic later.

Reply:
@LianJilu Thanks for the info. That makes sense. Then I think extra option like `useColor2ForDoji` is not required but you have to check if the open and close value are the same.